src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs

```diff
@@ -45,8 +45,20 @@ internal class Converter
             testElement.Storage = rockSteadyTestCase.Source;
             testElement.Priority = GetPriority(rockSteadyTestCase);
             testElement.Owner = GetOwner(rockSteadyTestCase);
+            testElement.Description = GetDescription(rockSteadyTestCase);
+            testElement.CssProjectStructure = GetCssProjectStructure(rockSteadyTestCase);
+            testElement.CssIteration = GetCssIteration(rockSteadyTestCase);
             testElement.ExecutionId = new TestExecId(executionId);
             testElement.ParentExecutionId = new TestExecId(parentExecutionId);
+            testElement.WorkItemIds = GetWorkItemIds(rockSteadyTestCase);
+
+            foreach (var trait in rockSteadyTestCase.Traits)
```

Comment:
Description, WorkItem, CssIteration and CssProjectStructure, are they part of the Traits collection ?

Reply:
They are all traits - however in the TRX report they have their own specific locations.

Reply:
@dotMorten These are specific to MSTest framework, aren't they ? If yes, can we try avoid these hardcoded checks. 
Also, these have not been part of the Traits on MSTestv1, but started showing up in MSTestv2 only and we have already fixed that issue in the adapter.

Reply:
> can we try avoid these hardcoded checks

These values were already hardcoded (apart from the new ones I added). Look for instance in the `GetPriority(…)` and `GetOwner(…)` methods.

> hese have not been part of the Traits on MSTestv1, but started showing up in MSTestv2 only and we have already fixed that issue in the adapter.

I'm not sure what you're getting at here or what you want me to do?

Reply:
- Principle we are trying to follow is TestPlatform should have minimal MSTest framework specific code.
- These properties (Description, WorkItem, CssIteration and CssProjectStructure) should not have been part of the Traits collection, that was a bug in the adapter and has been fixed.
- I really like the idea of Testproperty name-value available in the Trx. For these properties, I'd rather like them to be available as name value pairs like any other test property.
- I agree we have Priority and Owner as first class nodes, but these are legacy and have been there in older framework (backward compatibility).
- Considering Description for first class node still does make sense to me, as it can be extended to other testing frameworks.

Reply:
I'm not clear on how these are MSTest specific in any way. The trait names are rather generic. However there's an existing xml schema that defines all these properties as specific non-trait fields. These aren't the only properties that are pulled out from the traits collection and saved to the TRX in their own fields. I'm following the TRX schema 1:1 here. If another test framework wants to use for instance "Description" all they have to do is report a trait as "Description".
Whether non-traits should have been stored as traits in the first place is a moot point - that ship has already sailed, and this PR merely builds on what is already there and couldn't really change without breaking anyone.

> These properties (Description, WorkItem, CssIteration and CssProjectStructure) should not have been part of the Traits collection, that was a bug in the adapter and has been fixed

As I had [pointed out](https://github.com/Microsoft/testfx/pull/482#issuecomment-427454422) in the PR related to that, that change was a significant breaking change that I'm assuming you'll be rolling back before the next release before you break anyone. This specific PR was done this way to avoid changing any existing behavior and follow the exact patterns that were already established, so I have a hard time understand what the problem here is.

Reply:
Thanks for the explanation. I think I now agree with the approach, given the schema allows for it.
My main concern was around use of Traits to get these properties.
But the fact that these properties are available only through traits and not as part of first class properties in the TestCase, leaves us with no other choice but to go ahead with your approach.

On a side note, I am currently exploring changes that could help define how Traits should be used and if we need to have another property in the Trait class to be able to hide a properties when it comes to showing them the IDE. I will  revert the change https://github.com/Microsoft/testfx/pull/482 as part of this proposal.

Reply:
I have raised a PR for these test properties to be available in the TestCase object. Here is the PR: https://github.com/Microsoft/testfx/pull/538